spec/requests/vms_spec.rb

```diff
@@ -39,6 +39,25 @@ def update_raw_power_state(state, *vms)
     end
   end
 
+  context 'Vm index' do
+    # Preload Vm records
+    before do
+      @_vms = [vm, vm1, vm2, vm_openstack, vm_openstack1, vm_openstack2]
```

Comment:
Can you add a preload here for other records from the same table that _won't_ be in the query results?  For example, VMs that aren't in RBAC and perhaps a template? (Same table, different model).  Asking about teplates specifically because you chose `base_model` over `base_class`, which I assume was for a specific reason.

Reply:
> teplates

so close to tempates

Reply:
Fixed 😉

Reply:
> Asking about teplates specifically because you chose base_model over base_class, which I assume was for a specific reason.

I tested using both `base_model` and `base_class`, and there was still an N+1 with `base_class`.  But I can try again with this spec now, and in doing so, it fails to prevent `Rbac.filtered_object` from being called.

Reply:
> Fixed 😉

Not before I could quote it above.  Muhahahahaha!

Reply:
@NickLaMuro Ok, that's good..still think we should have specs that cover bonus vms as well as bonus templates. WDYT?

Reply:
@Fryguy I think I have this covered, but let me know what you think.